### PR TITLE
Initial differential query

### PR DESCRIPTION
**Pull request by groob**

### CODE REVIEW

**Comment by vivekuppal** [comment]
https://app.clubhouse.io/kolide/story/4537/built-in-queries-take-a-while-to-populate-from-the-launcher

File: pkg/osquery/extension.go [hunk 1]
```diff
@@ -144,12 +155,19 @@ func NewExtension(client service.KolideService, db *bolt.DB, opts ExtensionOpts)
 		return nil, errors.Wrap(err, "creating DB buckets")
 	}
 
+	identifier, err := IdentifierFromDB(db)
+	if err != nil {
+		return nil, errors.Wrap(err, "get host identifier from db when creating new extension")
+	}
+	initialRunner := &initialRunner{identifier: identifier, db: db, enabled: opts.RunDifferentialQueriesImmediately}
```
Comment (NickTitle) [inline pkg/osquery/extension.go]: you could skip this whole instantiation if you check the bool first - not sure if that matters

Comment (groob) [reply pkg/osquery/extension.go]: it matters because we still want to cache queries on first seen.

Comment (NickTitle) [reply pkg/osquery/extension.go]: smart, makes sense!

**Pull request merged**